File: front_end/src/app/Services/adherant.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { Adherant } from '../Models/Adherant';
import { Assurance } from '../Models/Assurance';
import { Beneficiaire } from '../Models/Beneficiaire';

@Injectable({
  providedIn: 'root'
})
export class AdherantService {

  private baseUrl = 'http://localhost:8034/api/adherant' ;


  constructor(private http: HttpClient) { }

  findbyAssuranceandMAtricule(assuranceNom: string , matricule: string): Observable<Adherant> {
    return this.http.get<Adherant>(`${this.baseUrl}/findbyAssuranceandMAtricule/${assuranceNom}/${matricule}`, {});
  }

  getAllAssuranceNames(): Observable<string[]> {
    return this.http.get<string[]>(`${this.baseUrl}/getAllAssuranceNames`);
  }

    ajouterAdherant(adherant: Adherant, idAssurance: number): Observable<Adherant> {
      return this.http.post<Adherant>(`${this.baseUrl}/${idAssurance}`, adherant);
    }

    getAdherantById(id: number): Observable<Adherant> {
      return this.http.get<Adherant>(`${this.baseUrl}/${id}`);
    }

    getBeneficiaireById(id: number): Observable<Beneficiaire> {
      return this.http.get<Beneficiaire>(`${this.baseUrl}/beneficiaire/${id}`);
    }

    getAllAdherants(): Observable<Adherant[]> {
      return this.http.get<Adherant[]>(`${this.baseUrl}/`);
    }

    modifierAdherent(id: number, nouvelAdherent: Adherant, assuranceId: number): Observable<Adherant> {
      return this.http.put<Adherant>(`${this.baseUrl}/${id}?assuranceId=${assuranceId}`, nouvelAdherent);
    }

    supprimerAdherant(id: number): Observable<void> {
      return this.http.delete<void>(`${this.baseUrl}/${id}`);
    }


}

File: backend/src/main/java/com/example/projectPfe/services/Interface/AdherantService.java
package com.example.projectPfe.Services.Interface;

import com.example.projectPfe.models.Adherant;
import com.example.projectPfe.models.Beneficiaire;
import com.example.projectPfe.models.Utilisateur;

import java.util.List;
import java.util.Optional;

public interface AdherantService {

    Adherant findAdherantByAssuranceNomAndMatricule(String assuranceNom, String matricule);

    List<String> getAllAssuranceNames();

    Adherant ajouterAdherant(Adherant adherant, int assurance);

    Optional<Adherant> getAdherantById(int id);

    Optional<Beneficiaire> getBeneficiaireById(int id);
    List<Adherant> getAllAdherants();
    Adherant modifierAdherent(int id, Adherant nouvelAdherent, int idAssurance);
    void supprimerAdherant(int id);

    Adherant desactiverAdherant(int adherantId);

    Adherant activerAdherant(int adherantId );
}
